core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -200,7 +200,7 @@ private[spark] class BlockManager(
     val shuffleConfig = new ExecutorShuffleInfo(
       diskBlockManager.localDirs.map(_.toString),
       diskBlockManager.subDirsPerLocalDir,
-      shuffleManager.getClass.getName)
+      conf.get("spark.shuffle.manager", "sort").toLowerCase)
```

Comment:
looks OK, but why not add a field in `ShuffleManager` itself that sets this? So you can just do `shuffleManager.name` or something here